mi/dataset/parser/pco2a_a_sample.py

```diff
@@ -0,0 +1,268 @@
+SENSOR_DATA_PATTERN += FLOAT  # supply voltage
+SENSOR_DATA_PATTERN += END_OF_LINE_REGEX
+
+SENSOR_DATA_PATTERN_AIR = TIMESTAMP + SPACE_REGEX  # dcl controller timestamp
+SENSOR_DATA_PATTERN_AIR += CHAR_A + SPACE_REGEX
+SENSOR_DATA_PATTERN_AIR += SENSOR_DATA_PATTERN
+SENSOR_DATA_MATCHER_AIR = re.compile(SENSOR_DATA_PATTERN_AIR)
+
+SENSOR_DATA_PATTERN_WATER = TIMESTAMP + SPACE_REGEX  # dcl controller timestamp
+SENSOR_DATA_PATTERN_WATER += CHAR_W + SPACE_REGEX
+SENSOR_DATA_PATTERN_WATER += SENSOR_DATA_PATTERN
+SENSOR_DATA_MATCHER_WATER = re.compile(SENSOR_DATA_PATTERN_WATER)
+
+print(SENSOR_DATA_PATTERN_WATER)
+
+# Manual test is below
+'''
+    from mi.dataset.parser.pco2a_a_dcl_new import SENSOR_DATA_MATCHER_WATER
+    data = "2019/02/27 00:00:09.290 W M,2019,02,27,00,00,11,44027,41190,446.74,40.60,11.37,28.52,1021,40.30,40.90,11.7\r\n"
+    m = SENSOR_DATA_MATCHER_WATER.match(data)
+    print(m.groups())
+'''
+
+# SENSOR_DATA_MATCHER produces the following groups.
+# The following are indices into groups() produced by SENSOR_DATA_MATCHER.
+# i.e, match.groups()[INDEX]
+SENSOR_GROUP_SENSOR_DATE_TIME = 9
+SENSOR_GROUP_ZERO_A2D = 10
+SENSOR_GROUP_CURRENT_A2D = 11
+SENSOR_GROUP_CO2 = 12
+SENSOR_GROUP_AVG_IRGA_TEMP = 13
+SENSOR_GROUP_HUMIDITY = 14
+SENSOR_GROUP_HUMIDITY_TEMP = 15
+SENSOR_GROUP_STREAM_PRESSURE = 16
+SENSOR_GROUP_DETECTOR_TEMP = 17
+SENSOR_GROUP_SOURCE_TEMP = 18
+SENSOR_GROUP_SUPPLY_VOLTAGE = 19
+SENSOR_GROUP_SAMPLE_TYPE = 8
+
+INSTRUMENT_PARTICLE_AIR_MAP = [
+    ('zero_a2d', SENSOR_GROUP_ZERO_A2D, int),
+    ('current_a2d', SENSOR_GROUP_CURRENT_A2D, int),
+    ('measured_air_co2', SENSOR_GROUP_CO2, float),
+    ('avg_irga_temperature', SENSOR_GROUP_AVG_IRGA_TEMP, float),
+    ('humidity', SENSOR_GROUP_HUMIDITY, float),
+    ('humidity_temperature', SENSOR_GROUP_HUMIDITY_TEMP, float),
+    ('gas_stream_pressure', SENSOR_GROUP_STREAM_PRESSURE, int),
+    ('irga_detector_temperature', SENSOR_GROUP_DETECTOR_TEMP, float),
+    ('irga_source_temperature', SENSOR_GROUP_SOURCE_TEMP, float),
+    ('supply_voltage', SENSOR_GROUP_SUPPLY_VOLTAGE, float)
+]
+
+INSTRUMENT_PARTICLE_WATER_MAP = [
+    ('zero_a2d', SENSOR_GROUP_ZERO_A2D, int),
+    ('current_a2d', SENSOR_GROUP_CURRENT_A2D, int),
+    ('measured_water_co2', SENSOR_GROUP_CO2, float),
+    ('avg_irga_temperature', SENSOR_GROUP_AVG_IRGA_TEMP, float),
+    ('humidity', SENSOR_GROUP_HUMIDITY, float),
+    ('humidity_temperature', SENSOR_GROUP_HUMIDITY_TEMP, float),
+    ('gas_stream_pressure', SENSOR_GROUP_STREAM_PRESSURE, int),
+    ('irga_detector_temperature', SENSOR_GROUP_DETECTOR_TEMP, float),
+    ('irga_source_temperature', SENSOR_GROUP_SOURCE_TEMP, float),
+    ('supply_voltage', SENSOR_GROUP_SUPPLY_VOLTAGE, float)
+]
+
+
+class DataParticleType(BaseEnum):
+    PCO2A_INSTRUMENT_AIR_PARTICLE = 'pco2a_a_dcl_instrument_air'
+    PCO2A_INSTRUMENT_WATER_PARTICLE = 'pco2a_a_dcl_instrument_water'
+#   PCO2A_INSTRUMENT_AIR_RECOVERED_PARTICLE = 'pco2a_a_dcl_instrument_air_recovered'
```

Comment:
ok to remove - maybe put in a comment that the recovered streams will use the same stream name if you want.

Reply:
Done